source/Halibut.Tests/Transport/Streams/CallCountingStream.cs

```diff
@@ -0,0 +1,185 @@
+﻿#nullable enable
+using System;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Halibut.Transport.Streams;
+#if NETFRAMEWORK
+using System.Runtime.Remoting;
+#endif
+
+namespace Halibut.Tests.Transport.Streams
+{
+    class CallCountingStream : AsyncDisposableStream
```

Comment:
Override all the methods! Again!

Another candidate to slim down when we have just the things you need to override as part of https://app.shortcut.com/octopusdeploy/story/57119/make-all-streams-async-friendly-you-can-t-mess-it-up

Reply:
This can be done if we either don't count calls to the old style async BegineRead/end methods.

or if the async stream no longer marks those streams as sealed:
![image](https://github.com/OctopusDeploy/Halibut/assets/7076477/1a97ebde-3d4e-4f58-9645-24eea3191248)

They are marked sealed on the assumption no one will ever need to care about them